plugins/plugin-chart-echarts/src/Pie/EchartsPie.tsx

```diff
@@ -16,10 +16,67 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import React from 'react';
-import { EchartsProps } from '../types';
+import { ensureIsArray } from '@superset-ui/core';
+import React, { useState, useEffect } from 'react';
+import { PieChartTransformedProps } from './types';
 import Echart from '../components/Echart';
+import { EventHandlers } from '../types';
 
-export default function EchartsPie({ height, width, echartOptions }: EchartsProps) {
-  return <Echart height={height} width={width} echartOptions={echartOptions} />;
+export default function EchartsPie({
+  height,
+  width,
+  echartOptions,
+  setDataMask,
+  labelMap,
+  groupby,
+}: PieChartTransformedProps) {
+  const [selectedValues, setSelectedValues] = useState<string[]>([]);
+
+  useEffect(() => {
+    // TODO: for now only process first selection - add support for nested
+    //  ANDs in ORs to enable multiple selection
+    const [groupbyValues] = selectedValues.map(value => labelMap[value]);
+    if (selectedValues.length === 0) return;
+    setDataMask({
+      crossFilters: {
+        extraFormData: {
+          append_form_data: {
+            filters: groupby.map((col, idx) => {
+              const val = groupbyValues[idx];
+              if (val === null || val === undefined)
+                return {
+                  col,
+                  op: 'IS NULL',
+                };
+              return {
+                col,
+                op: '==',
+                val: val as string | number | boolean,
+              };
+            }),
+          },
+        },
+        currentState: {
+          value: groupbyValues ?? null,
```

Comment:
also need to take current value from outside and set it to the chart as in `SelectFilterPlugin`